### PR TITLE
Add users option to feature toggle

### PR DESCRIPTION
**Pull request by joofsh**
- Support toggling a feature for individual users by css_id

Features supported:
- Enabling a feature by RO `FeatureToggle.enable!(:feature_name, regional_offices: ["RO33"])`
- Enabling a feature by individual user `FeatureToggle.enable!(:feature, users: ["CSS_ID"])`
- Enabling a feature for an RO and individual (non-RO) user: `FeatureToggle.enable!(:feature, users: ["CSS_ID"], regional_offices: ["RO33"])`

### CODE REVIEW

File: app/services/feature_toggle.rb [hunk 1]
```diff
@@ -19,15 +20,19 @@ def self.enable!(feature, regional_offices: nil)
...
   end
 
   # Method to disable a feature globally or for a specfic set of regional offices
   # Examples:
   # FeatureToggle.disable!(:foo)
   # FeatureToggle.disable!(:bar, regional_offices: ["RO01", "RO02"])
-  def self.disable!(feature, regional_offices: nil)
-    unless regional_offices
+  # FeatureToggle.disable!(:bar, users: ["CSS_ID_1", "CSS_ID_2"])
+  def self.disable!(feature, regional_offices: nil, users: nil)
```
Comment (amprokop) [inline app/services/feature_toggle.rb]: ah actually it's just to turn off a feature that's already been enabled, i see

File: app/services/feature_toggle.rb [hunk 2]
```diff
@@ -10,7 +10,8 @@ def self.features
   # Examples:
```
Comment (amprokop) [inline app/services/feature_toggle.rb]: maybe add "or users" to the line above?

File: app/services/feature_toggle.rb [hunk 3]
```diff
@@ -10,7 +10,8 @@ def self.features
...
   # FeatureToggle.enable!(:foo)
   # FeatureToggle.enable!(:bar, regional_offices: ["RO01", "RO02"])
-  def self.enable!(feature, regional_offices: nil)
+  # FeatureToggle.enable!(:bar, users: ["CSS_ID_1", "CSS_ID_2"])
+  def self.enable!(feature, regional_offices: nil, users: nil)
```
Comment (amprokop) [inline app/services/feature_toggle.rb]: hmm does `users` have to be the third parameter? so then is the example comment wrong?

File: app/services/feature_toggle.rb [hunk 4]
```diff
@@ -72,30 +84,61 @@ def self.redis
...
+    # is enabled for the user's ro
+    def enabled_for_regional_office?(regional_offices:, user:)
+      regional_offices.present? && user && regional_offices.include?(user.regional_office)
+    end
+
     def enable(feature:, key:, value:)
-      data = get_subkey(feature, key)
-      data = data.present? ? data + value : value
-      set_subkey(feature, key, data)
+      data = get_data(feature)
```
Comment (amprokop) [inline app/services/feature_toggle.rb]: looks good, if you feel like making these names a bit more descriptive that'd be cool (what data are we getting? like, would `get_currently_enabled_dict`  be more precise than `get_data` here?)

File: app/services/feature_toggle.rb [hunk 5]
```diff
@@ -72,30 +84,61 @@ def self.redis
...
-      data = feature_hash(feature)
-      data[key] if data
+    def get_data(feature)
+      feature_hash(feature) || {}
     end
 
-    def set_subkey(feature, key, data)
-      client.set(feature, { key => data.uniq }.to_json)
+    def set_data(feature, data)
+      client.set(feature, data.to_json)
```
Comment (amprokop) [inline app/services/feature_toggle.rb]: ah you moved it elsewhere nvm

File: app/services/feature_toggle.rb [hunk 6]
```diff
@@ -72,30 +84,61 @@ def self.redis
...
     def disable(feature:, key:, value:)
-      data = get_subkey(feature, key)
-      return unless data
-      set_subkey(feature, key, data - value)
+      return unless value
+
+      data = get_data(feature)
+      return unless data[key]
+
+      data[key] = data[key] - value
```
Comment (amprokop) [inline app/services/feature_toggle.rb]: you can do this?? crazy

File: app/services/feature_toggle.rb (hunk 3 above)
Comment (joofsh) [inline app/services/feature_toggle.rb]: it's using ruby keyword arguments, so either order works. 
`FeatureToggle.enable!(:test, users: [], regional_offices: [])` and `FeatureToggle.enable!(:test, regional_offices: [], users: [])` are both valid

File: app/services/feature_toggle.rb (hunk 4 above)
Comment (joofsh) [inline app/services/feature_toggle.rb]: went with `feature_enabled_hash`, which merges the `feature_hash` method anya already had with the `get_data` one

**Pull request merged**